pkg/skaffold/runner/v1/new.go

```diff
@@ -88,9 +89,10 @@ func NewForConfig(runCtx *runcontext.RunContext) (*SkaffoldRunner, error) {
 	podSelectors := kubernetes.NewImageList()
 
 	logger := getLogger(runCtx, kubectlCLI, podSelectors)
+	previewer := getResourcePreviewer(runCtx, kubectlCLI, podSelectors, labeller.RunIDSelector())
```

Comment:
From the description, it seems like deployer will provide the `previewer`. Does it makes sense to move the `ResourcePreviewer` creation to `getDeployer`?

In future, `deploy.kubectl.New`, `deploy.kustomize.New`, `deploy.helm.New` should be creating the `ResourcePreviewer` right?

Reply:
yeah definitely. I'll be addressing this immediately after merging this PR as per https://github.com/GoogleContainerTools/skaffold/pull/5809#discussion_r629621227

Reply:
I figured it would make that change easier to make/review if I kept it out of this PR

Reply:
sounds good!